client/db/src/light.rs

```diff
@@ -58,33 +59,45 @@ const HEADER_CHT_PREFIX: u8 = 0;
 /// Prefix for changes tries roots CHT.
 const CHANGES_TRIE_CHT_PREFIX: u8 = 1;
 
+/// Aux key to store pending pruning CHT ranges.
+const AUX_PENDING_CHT_RANGES: &'static[u8] = b"pending_cht_pruning_ranges";
+
 /// Light blockchain storage. Stores most recent headers + CHTs for older headers.
 /// Locks order: meta, cache.
 pub struct LightStorage<Block: BlockT> {
 	db: Arc<dyn Database<DbHash>>,
 	meta: RwLock<Meta<NumberFor<Block>, Block::Hash>>,
 	cache: Arc<DbCacheSync<Block>>,
 	header_metadata_cache: Arc<HeaderMetadataCache<Block>>,
+	shared_pruning_requirements: SharedPruningRequirementsSource<Block>,
+	pending_cht_pruning: RwLock<VecDeque<(NumberFor<Block>, NumberFor<Block>)>>,
```

Comment:
Actually we may only store number of the oldest block we want to prune. I.e. instead of `vec![(1, 2048), (2049, 4096), (4097, 5120), ..., (5121, 8192)]` we may only store single number - `8192` :)